tests/Sharpliner.Tests/AzureDevOps/ParameterReferenceTests.cs

```diff
@@ -32,16 +39,16 @@ private class ParameterReferenceTest_Template : StageTemplateDefinition
                 {
                     new Job("Job_1")
                     {
-                        Pool = parameters["pool"],
+                        Pool = parameters[p_pool],
```

Comment:
It's a step forward but I'd say you'd rather expect to pass `p_pool` only, right? As that's a parameter already.

Reply:
Definitely!

Reply:
So if we put this onto `Variable`:
```
public static implicit operator Conditioned<Pool>(Parameter parameter)
	=> new ParameterReference(parameter.Name); // Possibly something else
```

It just works out of the box. Technically we might need to use some other overload of `Conditioned<T>` but basically this is something that we should consider. I am not sure how to make it so that we don't need to list all possible target types but if you think about it, a variable should be possible to be inserted just at every place like this

Reply:
Are we able to go all the way and allow passing the parameters directly?

Reply:
Trying that here #392